### ISSUE TITLE
Remove need for glob import

### ISSUE THREAD

**Issue by kellpossible** [plausiblelabs/lens-rs#7]
Deriving `Lenses` on a type requires the import of a large number of items from `pl_lense` the suggested fix for this seems to be to use a glob import. I'm not a big fan of glob imports because they can break existing code even when adhering to semantic versioning, when a new type or function is introduced that clashes. Instead, perhaps using something like https://github.com/bkchr/proc-macro-crate or https://crates.io/crates/find-crate and importing the types in the derive macro might be a better approach?

**Comment by chrispcampbell** [comment]
@kellpossible Thanks for bringing this up.  The glob import has always bugged me.  Back when this library was first created (back in the pre-Rust 1.0 days) there wasn't any great solution, but support for `extern crate self as library_name` [was added](https://doc.rust-lang.org/reference/items/extern-crates.html) last year, which was helpful here.  I added that and fixed the macros to use fully qualified names.  The README now shows the minimal set of imports that are needed (only 2-4, depending on the use case).
